evalml/pipelines/components/estimators/regressors/arima_regressor.py

```diff
@@ -142,7 +142,25 @@ def _match_indices(self, X, y):
     def _set_forecast(self, X):
         from sktime.forecasting.base import ForecastingHorizon
 
-        fh_ = ForecastingHorizon([i + 1 for i in range(len(X))], is_relative=True)
+        # we can only calculate the difference if the indices are of the same type
```

Comment:
set this as a fallback to previous behavior in case we ever receive inconsistent indices

Reply:
Is the index type being different indicative of something?

Reply:
it should only believe the result of inconsistent user behavior and this is a safeguard against that!